### DISCUSSION TITLE
OpenAPI testing : schema not found for

### DISCUSSION THREAD

**Discussion by romch007** [adonisjs/core#4202] [Help]
I am trying to test an endpoint against OpenAPI spec. The spec part:
```yaml
paths:
  /groups/{id}:
    parameters:
      - in: path
        name: id
        schema:
          type: integer
        required: true
        description: ID of group
    delete:
      summary: Delete a group
      tags:
        - Groups
      security:
        - bearerAuth: []
      responses:
        '404':
           $ref: "#/components/responses/NotFoundError"
        '204':
          description: Group deleted successfully
```
The tests:
```ts
test('should delete a group', async ({ assert, client }) => {
  const group = await GroupFactory.create()

  const response = await client.delete(`/groups/${group.id}`).loginAs(adminUser)
  response.assertStatus(204)
  response.assertAgainstApiSpec()

  const deleteGroup = await Group.find(group.id)
  assert.isNull(deleteGroup)
})
```

When I run it I get: `Error: schema not found for {"path":"/groups/4","method":"delete","status":204}`
I have other tests and spec with the exact same path for method GET, POST and PUT that work fine, but this one just don't.

EDIT: OpenAPI assert fails on all endpoints that have a empty response body. [Link to related issue](https://github.com/PayU/api-contract-validator/issues/60).

**Comment by thetutlage** [comment]
Okay, so seems to be an issue with the underlying library (as you mentioned). So the best is to fix it with that library and update Japa dependencies.
